pulp_ansible/app/galaxy/v3/filters.py

```diff
@@ -0,0 +1,174 @@
+            ("-version", "by CV version (descending)"),
+        ),
+        fields={
+            "collection_version__pulp_created": "pulp_created",
+            "collection_version__namespace": "namespace",
+            "collection_version__name": "name",
+            "collection_version__version": "version",
+        },
+    )
+
+    def version_range_filter(self, queryset, name, value):
+        try:
+            s = semantic_version.SimpleSpec(value)
+            full_version_list = [
+                semantic_version.Version(v)
+                for v in queryset.values_list("collection_version__version", flat=True)
+            ]
+            version_list = [str(v) for v in s.filter(full_version_list)]
+
+            return queryset.filter(collection_version__version__in=version_list)
+        except ValueError:
+            raise ValidationError(_("%s must be a valid semantic version range." % name))
+
+    def filter_by_repository_name(self, queryset, name, value):
+        repository_names = self.request.query_params.getlist(name)
+        return queryset.filter(repository__name__in=repository_names)
+
+    def filter_by_repository_id(self, queryset, name, value):
+        repository_ids = self.request.query_params.getlist(name)
+        return queryset.filter(repository__pk__in=repository_ids)
+
+    def filter_by_distribution_id(self, qs, name, value):
+        dist_ids = self.request.query_params.getlist(name)
+
+        query = Q()
+
+        for dist_id in dist_ids:
+            for distribution in models.AnsibleDistribution.objects.filter(pk=dist_id):
+                if distribution.repository_version_id:
+                    query = query | Q(repository_version__pk=distribution.repository_version_id)
+                else:
+                    query = query | Q(repository__pk=distribution.repository_id)
+
+        return qs.filter(query)
+
+    def filter_by_base_path(self, qs, name, value):
+        base_paths = self.request.query_params.getlist(name)
+
+        query = Q()
+
+        for base_path in base_paths:
+            for distribution in models.AnsibleDistribution.objects.filter(base_path=base_path):
+                if distribution.repository_version_id:
+                    query = query | Q(repository_version__pk=distribution.repository_version_id)
+                else:
+                    query = query | Q(repository__pk=distribution.repository_id)
+
+        return qs.filter(query)
+
+    def repository_version_filter(self, qs, name, value):
+        if value != "latest":
+            return qs.filter(repository_version__number=value)
+
+        # Reduce queryset down to the "latest" repository version number(s)
+        rqs = (
+            RepositoryVersion.objects.filter(repository__pulp_type="ansible.ansible")
+            .order_by("repository", "-pulp_created")
+            .distinct("repository")
+            .values_list("repository__pk", "number")
+        )
+        query = Q()
```

Comment:
```suggestion
        query = Q(repository_version=None)
```
I think this part got dropped when I mentioned the problem of distros pointing to repo-vers that happened to be latest.